service/github/github.go

```diff
@@ -90,29 +113,46 @@ func (g *PullRequest) Flush(ctx context.Context) error {
 }
 
 func (g *PullRequest) postAsReviewComment(ctx context.Context) error {
-	comments := make([]*github.DraftReviewComment, 0, len(g.postComments))
+	if g.fallbackToLog {
+		// we don't have permission to post a review comment.
+		// Fallback to GitHub Actions log as report.
+		for _, c := range g.postComments {
+			if err := g.logWriter.Post(ctx, c); err != nil {
+				return err
+			}
+		}
+		return g.logWriter.Flush(ctx)
+	}
+
+	postComments := g.postComments
+	g.postComments = nil
+	rawComments := make([]*reviewdog.Comment, 0, len(postComments))
+	plainComments := make([]*github.PullRequestComment, 0, len(postComments))
+	reviewComments := make([]*github.DraftReviewComment, 0, len(postComments))
 	remaining := make([]*reviewdog.Comment, 0)
-	for _, c := range g.postComments {
+	for _, c := range postComments {
 		if !c.Result.InDiffFile {
 			// GitHub Review API cannot report results outside diff. If it's running
-			// in GitHub Actions, fallback to GitHub Actions log as report .
+			// in GitHub Actions, fallback to GitHub Actions log as report.
 			if cienv.IsInGitHubAction() {
-				githubutils.ReportAsGitHubActionsLog(c.ToolName, "warning", c.Result.Diagnostic)
+				if err := g.logWriter.Post(ctx, c); err != nil {
+					return err
+				}
 			}
 			continue
 		}
 		body := buildBody(c)
 		if g.postedcs.IsPosted(c, githubCommentLine(c), body) {
+			// it's already posted. skip it.
 			continue
 		}
+
+		rawComments = append(rawComments, c)
 		if !c.Result.InDiffContext {
 			// If the result is outside of diff context, fallback to GitHub Review
 			// Comment API.
 			comment := buildPullRequestComment(c, body, g.sha)
```

Comment:
[Unrelated to this pr]
I noticed that line (position) information is completely lost for these file comments and it must be confusing for most cases. 🤔
Probably we should add line (position) information in comment body. 

(You don't need to address this comment in this pr. )